icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -1023,11 +1021,8 @@ DateIntervalFormat::getDateTimeSkeleton(const UnicodeString& skeleton,
     }
 
     /* generate normalized form for time */
-    if ( HCount != 0 ) {
-        normalizedTimeSkeleton.append(CAP_H);
-    }
-    else if ( hCount != 0 ) {
-        normalizedTimeSkeleton.append(LOW_H);
+    if (hourChar != u'\0' ) {
+        normalizedTimeSkeleton.append(hourChar);
```

Comment:
I think we actually _couldn't_ do it in `normalizedTimeSkeleton`, because I think if we changed it there, we'd lose the fact that the original skeleton had a `K` in it instead of an `h`.  I may be mixed up, but I think the original version of the skeleton is how we do this, and that the only thing normalization does is mess with the field lengths.  But I'm coming to this code fairly cold, so I might not know all the history.  As you point out, though, this is how we were already handling `v` and `z`, so I just piggybacked on top of that.

Reply:
I think you are right.